### PR TITLE
fix(Settings): use schema

### LINKED ISSUE #2237: Allow to set output format in CLI
**Is your feature request related to a problem? Please describe.**

Running the following statement

```sh
$ echo 'c = \pm\sqrt{a^2 + b^2}' | npx katex
```

renders this HTML fragment:

```html
<span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>c</
mi><mo>=</mo><mo>±</mo><msqrt><mrow><msup><mi>a</mi><mn>2</mn></msup><mo>+</mo><msup><mi>b</mi><mn>2</mn></msup></mrow><
/msqrt></mrow><annotation encoding="application/x-tex">c = \pm\sqrt{a^2 + b^2}
</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut
" style="height:0.43056em;vertical-align:0em;"></span><span class="mord mathdefault">c</span><span class="mspace" style=
"margin-right:0.2777777777777778em;"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.277777
7777777778em;"></span></span><span class="base"><span class="strut" style="height:1.04em;vertical-align:-0.1266110000000
0003em;"></span><span class="mord">±</span><span class="mord sqrt"><span class="vlist-t vlist-t2"><span class="vlist-r">
<span class="vlist" style="height:0.913389em;"><span class="svg-align" style="top:-3em;"><span class="pstrut" style="hei
ght:3em;"></span><span class="mord" style="padding-left:0.833em;"><span class="mord"><span class="mord mathdefault">a</s
pan><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.740108em;"><sp
an style="top:-2.9890000000000003em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class=
"sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span><sp
an class="mspace" style="margin-right:0.2222222222222222em;"></span><span class="mbin">+</span><span class="mspace" styl
e="margin-right:0.2222222222222222em;"></span><span class="mord"><span class="mord mathdefault">b</span><span class="msu
psub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:0.740108em;"><span style="top:-2.989
0000000000003em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6
size3 mtight"><span class="mord mtight">2</span></span></span></span></span></span></span></span></span></span><span sty
le="top:-2.873389em;"><span class="pstrut" style="height:3em;"></span><span class="hide-tail" style="min-width:0.853em;h
eight:1.08em;"><svg width='400em' height='1.08em' viewBox='0 0 400000 1080' preserveAspectRatio='xMinYMin slice'><path d
='M95,702
c-2.7,0,-7.17,-2.7,-13.5,-8c-5.8,-5.3,-9.5,-10,-9.5,-14
c0,-2,0.3,-3.3,1,-4c1.3,-2.7,23.83,-20.7,67.5,-54
c44.2,-33.3,65.8,-50.3,66.5,-51c1.3,-1.3,3,-2,5,-2c4.7,0,8.7,3.3,12,10
s173,378,173,378c0.7,0,35.3,-71,104,-213c68.7,-142,137.5,-285,206.5,-429
c69,-144,104.5,-217.7,106.5,-221
l0 -0
c5.3,-9.3,12,-14,20,-14
H400000v40H845.2724
s-225.272,467,-225.272,467s-235,486,-235,486c-2.7,4.7,-9,7,-19,7
c-6,0,-10,-1,-12,-3s-194,-422,-194,-422s-65,47,-65,47z
M834 80h400000v40h-400000z'/></svg></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span
 class="vlist" style="height:0.12661100000000003em;"><span></span></span></span></span></span></span></span></span>
```

This contains both, the MathML and HTML output.
Firefox (75) renders it like this:

katex-mathml-and-html

The first equation is rendered correctly, and the second one should not be visible.

**Describe the solution you'd like:**

Allow to set the _output_ format, as described in the [options documentation](https://katex.org/docs/options.html), through the CLI, e.g. `katex --output-format html`.

### PR DESCRIPTION
**Pull request by ylemkimon**
**What is the previous behavior before this PR?**

The options of `Settings` were repeated in `cli.js`, `type SettingsOptions`, `class Settings`, and `Settings::constructor()`. This caused some options missing in the CLI, #2237, #2428, and `minRuleThickness`.

**What is the new behavior after this PR?**

DRY options in `cli.js`, `Settings::constructor()`, and `type SettingsOptions`. Use `SETTINGS_SCHEMA` and `$Shape<Settings>`, respectively.

This adds `-f, --format <type>` and `--min-rule-thickness <size>` to the CLI. Fixes #2237 and closes #2241. It also exports `SETTINGS_SCHEMA`, so developers and users could use.

Currently, Flow checks if a non-existent property on Settings exists on the schema.

In the future, we could
- validate `options` against the schema, throwing on unknown values or type mismatches (this would be a breaking change)
- automatically generate Options docs from the schema
- check all properties on `Settings` exist on the schema
- check the type of a property on `Settings` matches the schema (I'd like to do these two after #2110)

### CODE REVIEW

File: src/Settings.js [hunk 1]
```diff
@@ -52,23 +52,180 @@ export type TrustContextTypes = {
...
+        type: "boolean",
+        description: "Render math in display mode, which puts the math in " +
+            "display style (so \\int and \\sum are large, for example), and " +
+            "centers the math on the page on its own line.",
+        cli: "-d, --display-mode",
+    },
+    output: {
+        type: {enum: ["htmlAndMathml", "html", "mathml"]},
+        description: "Determines the markup language of the output.",
+        cli: "-f, --format <type>",
```
Comment (edemaine) [inline src/Settings.js:117]: `-f` conflicts with `--macro-file`.  I guess we should remove it for backward compatibility?

**Review by edemaine — COMMENTED** [review]
Wow, this is an impressive rewrite; it shows the extensive complexity of our options system.

I did a bunch of testing of cli (should we have automated tests for this??), and found one bug/change in the behavior of `-f`. Otherwise looks good.

File: src/Settings.js (hunk 1 above)
Comment (ylemkimon) [reply src/Settings.js:117]: Let's go with `-F`.

**Comment by ylemkimon** [comment]
> should we have automated tests for this?

Tracked in #1503.

**Review by edemaine — APPROVED** [review]
Thanks for all your recent PRs!

**Comment by ylemkimon** [comment]
Thank you for all the reviews!

**Pull request merged**